src/main/java/picard/sam/SamToFastq.java

```diff
@@ -466,40 +474,51 @@ private void assertPairedMates(final SAMRecord record1, final SAMRecord record2)
      * Allows for lazy construction of the second-of-pair writer, since when we are in the "output per read group mode", we only wish to
      * generate a second-of-pair fastq if we encounter a second-of-pair read.
      */
-    static final class FastqWriters {
+    private static final class FastqWriters {
         private final FastqWriter firstOfPair, unpaired;
         private final Lazy<FastqWriter> secondOfPair;
+        private final Map<SAMReadGroupRecord, List<FastqWriter>> additionalWriters;

```

Comment:
This field is in an object that itself is in a `Map<SAMReadGroupRecord, FastqWriters>`. Do we need a map inside a map for this feature? It seems like what you really want is `writers` and `additionalWriters` as both being local variables in `doWork()`.

Reply:
nice catch, I split it out into its own variable in `doWork()` for now and we can change after we talk